posthog/api/person.py

```diff
@@ -508,6 +572,52 @@ def cohorts(self, request: request.Request) -> response.Response:
 
         return response.Response({"results": CohortSerializer(cohorts, many=True).data})
 
+    @action(methods=["GET"], url_path="activity", detail=False)
+    def all_activity(self, request: request.Request, **kwargs):
+        activity = load_activity(scope="Person", team_id=self.team_id)
+        return Response(
+            {"results": ActivityLogSerializer(activity, many=True,).data, "next": None, "previous": None,},
+            status=status.HTTP_200_OK,
+        )
+
+    @action(methods=["GET"], detail=True)
+    def activity(self, request: request.Request, **kwargs):
+        item_id = kwargs["pk"]
+        if not self.get_queryset().filter(id=item_id, team_id=self.team_id).exists():
+            return Response("", status=status.HTTP_404_NOT_FOUND)
+
+        activity = load_activity(scope="Person", team_id=self.team_id, item_id=item_id,)
+        return Response(
+            {"results": ActivityLogSerializer(activity, many=True,).data, "next": None, "previous": None,},
+            status=status.HTTP_200_OK,
+        )
```

Comment:
These two actions seem quite boilerplate-y, having just reviewed the PR around feature flags. Could they go into some mixin? Looks like `scope` is the only customizable part.

Reply:
Hmmm 🤔 

There's the existence check too... although possible that could be removed now there's no history hog (RIP)

They are boilerplate, but also, only two examples currently and more coming. So I'd suggest we say "a little duplication is better than the wrong abstraction" and push this decision into the Insights implementation of this

Reply:
Sure, _Rule of Three_!